convert2rhel/unit_tests/special_cases_test.py

```diff
@@ -0,0 +1,80 @@
+from unittest import mock
+
+import pytest
+
+from convert2rhel import special_cases
+from convert2rhel.special_cases import (
+    OPENJDK_RPM_STATE_DIR,
+    perform_java_openjdk_workaround,
+)
+
+
+@pytest.mark.parametrize(
+    (
+        "has_openjdk",
+        "can_successfully_apply_workaround",
+        "mkdir_p_should_raise",
+        "check_message_in_log",
+        "check_message_not_in_log",
+    ),
+    [
+        # All is fine case
+        (
+            True,
+            True,
+            None,
+            "openjdk workaround applied successfully.",
+            "Can't create %s" % OPENJDK_RPM_STATE_DIR,
+        ),
+        # openjdk presented, but OSError when trying to apply workaround
+        (
+            True,
```

Comment:
Would this styling be possible? Makes it a bit easier to read when there's a lot of parameters
```suggestion
            has_openjdk=True,
```

Reply:
I don't know any better option. Technically it is possible to adopt the dict: Dict[str, List[Any]] as a parameterization arg, but I don't think it will improve the readability too much. Another option to make a table like crazy formatting:) But what I am usually doing here is controlling the number of parameterization params. So it should exceed 4-5 params, so I can keep them in memory:)